test/performance/shmem_perf_suite/shmem_bw_nb_atomics_perf.c

```diff
@@ -0,0 +1,283 @@
+/*
+ *  Copyright (c) 2018 Intel Corporation. All rights reserved.
+ *  This software is available to you under the BSD license below:
+ *
+ *      Redistribution and use in source and binary forms, with or
+ *      without modification, are permitted provided that the following
+ *      conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*
+**
+**  This is a bandwidth centric test for nb atomic operations
+**
+**  Features of Test: uni-directional bandwidth
+**
+**  -by default megabytes/second results
+**
+*/
+#include <bw_common.h>
+
+#define ATOMIC_COMM_STYLE COMM_INCAST
+
+#define uni_bw(len, metric_info, snode, NAME, TYPE, op)                        \
+    do {                                                                       \
+        double start = 0.0, end = 0.0;                                         \
+        unsigned long int i = 0, j = 0;                                        \
+        int dest = partner_node(metric_info);                                  \
+        TYPE fetch;                                                            \
+        shmem_barrier_all();                                                   \
+                                                                               \
+        switch(op) {                                                           \
+            case OP_FETCH:                                                     \
+                if(snode) {                                                    \
+                    for(i = 0; i < metric_info->warmup; i++) {                 \
+                        for(j = 0; j < metric_info->window_size; j++)          \
+                            shmemx_##NAME##_atomic_fetch_nbi((TYPE *) &fetch,  \
+                                (TYPE *)(metric_info->dest), dest);            \
+                            shmem_quiet();                                     \
```

Comment:
Indenting of the quiet looks incorrect?  Was scratching my head for a sec wondering why we would put it inside the window loop.  :)

Reply:
Sorry about that. Will correct it.